<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset='utf-8'>
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-dark.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>BB2491 Project by John Lamb</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>Biolamb</h1>
          <h2>BB2491 Project</h2>
          <h2>John Lamb</h2>
        </header>

        <section id="downloads" class="clearfix">
          <h2></h2>
          <a href="https://github.com/johnlamb/biolamb/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/johnlamb/biolamb/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/johnlamb/biolamb" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <h3><a id="eleven-entry" class="anchor" href="#eleven-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-12 Control peaks, full automation and improvements</h3>
        <p>The pipeline is improved as being fully automatic. Fastq file to peaks results file are now automated. A logical next step on this
          is to modularize the pipeline as well(right now the pipeline(s) are fixed tools). <br>
          Have run more experiments with Simons supplied GQP.bed file which shows the locations of the G4Q complexes. Have done them with all 4 variants
          of the pipeline and handed it over to the others. We will hopefully have some concrete biological answers early this week so we can
          discuss with Lars at this weeks seminar were we stand on material.<br><br>
          I have looked closer at our toolchain and the best option of the once we have tried seems to be Bowtie2 in combination with MACS2. No suprise there with
          Bowtie2 being a leader in short sequencing alignments and MACS2 being one of the popular peak callers. Important to not compare Macs (1.1) and MACS2
          as they calculate p-values and q-values differently. MACS(1.1) mainly uses p-value as cutoff and hence no family wise error correction
          while the newer MACS2 predominantly pusches for the family wise correction of q-value, especially as it also reports FDR.
          <br>
          Some improvments that could be done with more time (on top of the improvements already implemented in the pipeline)    
          </p>
           <b>Biological</b>
          <ul>
            <li>Repeat the sequencing with multiple samples. Each sample has only been sequenced once, should be at least 3, pref 5 for statistical rigour.</li>
            <li>Do a separate analysis on the G4Q locations on the same wildtypes as in the samples, H33 and WT26.</li>
          </ul>
          
            <b>Technical</b>
            <ul>
              <li>Do a grid search or the like for different parameters for the different tools. (Also evaluate more tools)</li>
              <li>Fully modularize the automation, which in effect means build a new program specifically for comparing peak locations with a reference</li>
            </ul>
          <h3><a id="ten-entry" class="anchor" href="#ten-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-09 Progress and pipeline</h3>
        <p>We had a productive meeting yesterday were we went through the first peak data. On initial analysis we have an increase in DSB in both the H33 KO and ATRX KO. The ATRX KO given a 5 fold increase
          while H33 KO a mere 50%. The DAXX showed a decline which raises alarm but might be a artifact of not running repeat experiments.<br>
          We are now evaluating the pipeline. We have run bwa-aln as the most widely used aligner but also tried bwa-mem. However with these short reads (60bp) we will most
          likely not see a difference. Weilong is doing a quality analysis for this. We have also tried running bowtie2 as the defact speed standard of the modern aligners. Remains to be seen if 
          it even matters for our project as the aligning in itself has less impact as long as it is over a certain threshold.<br>
          Thus far we have only run MACS2 as a peak caller. We do have an interest in trying Homer findPeaks and maybe SISSR as more moderna tools but time will
          tell if we will have to push it for a further project or include in this.<br>
          A good way to improve the tool chain at this stage is to fully automate the running, so we can give one (or more) fastq files and a reference and this will
          result in the peaks (which we are happy with as final result for the tool chain). The chain is almost fully automated now but I am
          in the process of finalizing it and also  make sure it is fully modular so you would easily be able to decide yourself what aligner and peak caller to use.
          
          
          </p>
          <h3><a id="ninth-entry" class="anchor" href="#ninth-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-07 Jobs for tomorrow</h3>
         <p>Good seminar which helped to see where we stand and what stands before us. Got some interesting info on the Molecular Identifier and Barcode in the BLISS
          protocol that these are usually used in the peak calling protocols as well. What we understood from Simon they are not needed for us as splitting of the samples
          and normalization has already been done. We are however checking this with Simon.<br><br>
          I have run MACS2 peak calling for the trimmed dataset and Weilong will be doing summary and descriptive statistics of the result before the meeting tomorrow.
          <br> We do have some concern about RNA 3' adapter contamination (as per the FastQC report) but after discussion with Lars we will start to run the trimmed
          dataset and see where that leaves us. Depending on the result and time this is otherwise a good step for improving the pipeline, at least confirming there is or isn't
          a change by cutting the (possibly) contaminated reads out.<br><br>
          Another aspect is to not overdo the pipeline. We are now running mapping with bwa-mem, bwa-aln and bowtie2 as these three are the main once for Illumina sequenced short (less than 100bp) single reads.
          Jing will be compare the bwa-aln and bwa-mem mapping to see what the difference is. With the short reads we have (60 bp) there will most likely not be any difference between mem/aln, or so our 
          experience tells us.<br><br>
          Leonard will be starting the poster work so we have a template as soon as we have more data, better early than late, and I will be expanding on our pipeline. The main thing now is to get the pipeline fully automated so we can run it quickly
          and to test the tools we have. <br>
          Apart from the mappers we are also looking at the peak callers, at present we hace run MACS2 as we have experience in it but we will also try Homer(findPeaks) as this is one of the for front tools. However, I need to inventories Uppmax and
          see what tools are easily available.<br><br>
          <b>Update! </b> Apart from just peak calling, we should also do differential peaks between the wildtypes and each respective sample, that way we should get the statistical power
          behind each extra peak.<br>
          <b>Update!</b> Leonard found something really interesting about the RNA primer contaminations and the "extra" sequences that
          was added after the primer to fill out the gap. Seems there has been a problem before and why they map to the carp genom:<br>
          
          
        <a href="http://grahametherington.blogspot.se/2014/09/why-you-should-qc-your-reads-and-your.html ">http://grahametherington.blogspot.se/2014/09/why-you-should-qc-your-reads-and-your.html</a>
          </p>
          
          
          
        
        <h3><a id="eight-entry" class="anchor" href="#eight-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-07 Peaks and delegation</h3>
         <p>Built a small pipeline for the peak calling using MACS2. Running it as we speak(read). <br>
          We also got the G4Q-complex locations BED files from Simon so can use those to see if there is any correlation between the
          peaks and G4Q-location. Will be interesting to see if there are any correlations.<br><br>
          We need to have a serious talk in the group today about the work that is done and delegation. We need to have all data ready 
          before the christmas break and people needs to step up and start prioritizing this project if we are to complete it in time.
          
          </p>
          
          <h3><a id="seven-entry" class="anchor" href="#seven-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-06 Pipeline</h3>
         <p>A full pipeline is set up and documented (commands.txt in the project root folder on Uppmax). This will take the (trimmed and quality checked) files
          and map (to .sai with bwa aln or .sam with bwa mem) and then convert/index/sort etc into .bed files ready for MACS and/or viewing.<br><br>
          Seminar tomorrow, questions we need to discuss/talk to Lars about:
          </p>
          <ul>
            <li>Preprocessing steps, have we missed something? RNA 3' adapters cleaning?</li>
            <li>Statistics, any idea of tests to explore?</li>
            <li>Tools pipeline, bwa aln/mem, samtools and MACS.</li>
            <li>Scope of project</li>
          </ul>
          
        
        
        
        <h3><a id="six-entry" class="anchor" href="#six-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-05 Project plan and QC</h3>
          <p>Good check up meeting. Clear path on what we all have to do for this week.<br><br>
          I trimmed the fastq files for the beginning MI and barcode, 16 bases, and ran fastqc again. Instantly looks better. However we seem to have som 3' Illumina
          adapter contamination. Leonard and I went through and checked and found the exact sequence. We will do some research on <i>cutadapt</i> to 
          remove them.<br><br>
          I am rerunning the mapping against the trimmed files, most likely using <b>bwa aln</b> instead of <b>bwa mem</b> as the sequences
          now are down to 60 bp. Will be interesting to see what happens with RNA primer purging.<br><br>
          Looking into peak calling, in the middle of a small script to make sure all the files are in proper format (bed) for MACS (and looking at
          least one more tool). Jing will ask Simon for the bed-files that indentifies the G4Q sites so we can correlate.<br><br>
          The updated project goal: <i>Are there a significant increase in DBS between the wildtype and knockout samples? Are these breaks correlated to the known G4Q sites?</i>
          <br><br>
          We need to get Lars input on the statistics on Wednesday, what would be the best approach to calculate the confidance given so few samples?
          </p>
        
        
          <h3><a id="fifth-entry" class="anchor" href="#fifth-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-04 Alignments</h3>
          <p>Sorted the mm10 mouse reference. The full one came in 2bit format so have done some conversion (straight forward) into .fa format.
          Built script and run the index (bwa) on the reference, reside in the data folder. Have also got a script up and running for alignment with bwa-mem (default parameters).
          Simple support for running all 5 samples in parallel. 
          <br><br>
          Went through the fastqc result and there are some question marks. Specifically 3'-adapters in the latter part of the reads and som
          strange nucleotide spreading in the first 3rd. We also have to look for the Molecular Identifier and Barcode patterns. Remember they are both
          8 base pairs long and the Barcode always ends with a C. Should be easy to run some quick statistics to verify the 4 diff positions (pos 8, 16 and -8 and -16). 
          <br><br>
          Meeting tomorrow at 1300. We need to have all quality questions sorted this week and preferably a full pipeline for the analysis done so we quickly can run everything through.
          
          </p>
        
        <h3>
<a id="fourth-entry" class="anchor" href="#fourth-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-02 Scripts</h3>

<p> Done some quick scripting on Uppmax. We have around 15-22 million reads in each file (with each read being 76bp, have to confirm this is the case for all samples).
          </p><ul id="sample_list">
    <li>H3.3 Wildtype   21,079,172 reads</li>
    <li>H3.3 KO         16,330,970 reads</li>
    </ul>
      <ul>
    <li>WT26 Wildtype   12,773,603 reads</li>
    <li>ATRX KO         24,615,902 reads</li>
    <li>DAXX KO         11,603,719 reads</li>
  </ul>
<p>
  We had a good meeting with Simon and have updated our research question. <br>
          <i> How many significant Double Stranded Breaks (peaks) is there in the KO samples compared to its respective wildtype (adjust for "freak peaks" by comparing the samples 
          against a paired-end sequencing of genomic DNA from wildtype ES cells at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1555118">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1555118</a>)
          and do these peaks (DSB) correspond to known location of the G4Q-complex? G4Q-complex in coding regions or non-coding regions?</i>
  <br>Upcoming group meetings:</p>
    <ul>
      <li>Monday 5th Dec, 13:00, Pluto Alpha 4</li>
      <li>Thursday 8th Dec, 10:00, Kinnekulle Alpha 5</li>
          </ul>
 
<hr>
          <h3>
<a id="third-entry" class="anchor" href="#third-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-01 Start the pipeline</h3>

<p>Project structure set up on Uppmax. A simple straight forward version, might be we set up a repo for the code so we all can share it 
  easily.<br>
  Trello set up for us all and a board each for both the pipeline work and the initial research started</p>
<p>Done some research on the initial pipeline. With the mouse genom being almost 3GB long (2.8x10^9) we need to choose an aligner suited for longer genomes. BWA-mem is a good
  fallback but SHRiMP2 and Novoalign seems like interesting options, especially SHRiMP2 which does not need indexing. Bowtie2 and STAR should also be
  in the run, requires quite a lot of ram (30+GB) but should not be a problem on Uppmax. <br>
  No requirement for splicing though which simplifies the alignment. Will be interesting to see what Simon says tomorrow. Might
  be he already has a pipeline in mind. Always easier to use what the group is already using.
          </p>
         
<hr>
           <h3>
<a id="second-entry" class="anchor" href="#second-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-11-30 Runthrough of project</h3>

<p>First seminar and runthrough of the project. I will set up a structure on Uppmax for us to use for code/data. </p>
          <p>As there are a lot of questions about the projects, specifically 
  what exact biological question we should focus on in this project, we have decided to book in a meeting with Jings supervisor (Simon Elsässer) to both get a good understanding
          of the new BLISS method and to get more information on what has been done previously (both using BLESS and BLISS). UPDATE! Meeting set for 1100 on friday 2 Dec.</p>
<hr>
          <h3>
<a id="first-entry" class="anchor" href="#first-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-11-29 Start of project</h3>

<p>This will be the project diary for the BB2491 project HT2016. Blog set up and data moved to Uppmax. All five(5) files reside at /proj/g2016025/Grp4Prj/BLISS/. 
          We have to look closer at the specifics. Each sample has only been run once and the Double Stranded Breaks (DSB) has been luminecently tagged. Will 
          be interesting to see how this presents itself in they data.</p>
          <p>Must have a closer look at the statistic approach. Without multi run per sample we need to make sure the statistics are valid and we run the right tests.</p>

 </section>

        <footer>
          Biolamb is maintained by <a href="https://github.com/johnlamb">johnlamb</a><br>
        </footer>

        
      </div>
    </div>
  </body>
</html>
