<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-dark.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>BB2491 Project by John Lamb</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>Biolamb</h1>
          <h2>BB2491 Project</h2>
          <h2>John Lamb</h2>
        </header>

        <section id="downloads" class="clearfix">
          <a href="https://github.com/johnlamb/biolamb/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/johnlamb/biolamb/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/johnlamb/biolamb" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <h3><a id="fifth-entry" class="anchor" href="#fifth-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-04 Alignments</h3>
          Sorted the mm10 mouse reference. The full one came in 2bit format so have done some conversion (straight forward) into .fa format.
          Built script and run the index (bwa) on the reference, reside in the data folder. Have also got a script up and running for alignment with bwa-mem(default parameters).
          Simple support for running all 5 samples in parallel. 
          <br>
          Went through the fastaqc result and there are some question marks. Specifically 3'-adapters in the latter part of the reads and som
          strange nucleotide spreading in the first 3rd. We also have to look for the Molecular Identifier and Barcode patterns. Remember they are both
          8 base pairs long and the Barcode always ends with a C. Should be easy to run some quick statistics to verify the 4 diff positions (pos 8, 16 and -8 and -16). 
          <br>
          Meeting tomorrow at 1300. We need to have all quality questions sorted this week and preferably a full pipeline for the analysis done so we quickly can run everything through.
          
          </p>
        
        <h3>
<a id="fourth-entry" class="anchor" href="#fourth-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-02 Scripts</h3>

<p> Done some quick scripting on Uppmax. We have around 15-22 million reads in each file (with each read being 76bp, have to confirm this is the case for all samples).
  <ul id="sample_list">
    <li>H3.3 Wildtype   21,079,172 reads</li>
    <li>H3.3 KO         16,330,970 reads</li>
    <br>
    <li>WT26 Wildtype   12,773,603 reads</li>
    <li>ATRX KO         24,615,902 reads</li>
    <li>DAXX KO         11,603,719 reads</li>
  </ul>

  We had a good meeting with Simon and have updated our research question. <br>
          <i> How many significant Double Stranded Breaks (peaks) is there in the KO samples compared to its respective wildtype (adjust for "freak peaks" by comparing the samples 
          against a paired-end sequencing of genomic DNA from wildtype ES cells at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1555118">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1555118</a>)
          and do these peaks (DSB) correspond to known location of the G4Q-complex? G4Q-complex in coding regions or non-coding regions?</i>
<br>Upcoming group meetings:
    <ul>
      <li>Monday 5th Dec, 13:00, Pluto Alpha 4</li>
      <li>Thursday 8th Dec, 10:00, Kinnekulle Alpha 5</li>
          </ul>
       
          </p>
 
<hr>
          <h3>
<a id="third-entry" class="anchor" href="#third-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-12-01 Start the pipeline</h3>

<p>Project structure set up on Uppmax. A simple straight forward version, might be we set up a repo for the code so we all can share it 
  easily.<br>
  Trello set up for us all and a board each for both the pipeline work and the initial research started</p>
<p>Done some research on the initial pipeline. With the mouse genom being almost 3GB long (2.8x10^9) we need to choose an aligner suited for longer genomes. BWA-mem is a good
  fallback but SHRiMP2 and Novoalign seems like interesting options, especially SHRiMP2 which does not need indexing. Bowtie2 and STAR should also be
  in the run, requires quite a lot of ram (30+GB) but should not be a problem on Uppmax. <br>
  No requirement for splicing though which simplifies the alignment. Will be interesting to see what Simon says tomorrow. Might
  be he already has a pipeline in mind. Always easier to use what the group is already using.
          </p>
         
<hr>
           <h3>
<a id="second-entry" class="anchor" href="#second-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-11-30 Runthrough of project</h3>

<p>First seminar and runthrough of the project. I will set up a structure on Uppmax for us to use for code/data. </p>
          <p>As there are a lot of questions about the projects, specifically 
  what exact biological question we should focus on in this project, we have decided to book in a meeting with Jings supervisor (Simon Els√§sser) to both get a good understanding
          of the new BLISS method and to get more information on what has been done previously (both using BLESS and BLISS). UPDATE! Meeting set for 1100 on friday 2 Dec.</p>
<hr>
          <h3>
<a id="first-entry" class="anchor" href="#first-entry" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>2016-11-29 Start of project</h3>

<p>This will be the project diary for the BB2491 project HT2016. Blog set up and data moved to Uppmax. All five(5) files reside at /proj/g2016025/Grp4Prj/BLISS/. 
          We have to look closer at the specifics. Each sample has only been run once and the Double Stranded Breaks (DSB) has been luminecently tagged. Will 
          be interesting to see how this presents itself in they data.</p>
          <p>Must have a closer look at the statistic approach. Without multi run per sample we need to make sure the statistics are valid and we run the right tests.</p>

 </section>

        <footer>
          Biolamb is maintained by <a href="https://github.com/johnlamb">johnlamb</a><br>
        </footer>

        
      </div>
    </div>
  </body>
</html>
